core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java

```diff
@@ -128,12 +129,23 @@ default void checkCanSetUser(Optional<Principal> principal, String userName)
      * Checks if identity can execute a query.
      *
      * @throws AccessDeniedException if not allowed
+     * @deprecated use {@link #checkCanExecuteQuery(Identity, QueryId)}
```

Comment:
add `@Deprecated` annotation. is allows IDEs to see deprecation even if source/javadoc is not downloaded, and is likely required by many tools to observe the deprecation (javac?)

Reply:
Forgot about it, thanks